lib/tokenVersions.js

```diff
@@ -0,0 +1,235 @@
+ */
+export async function get({id, tokenizerId}) {
+  const query = {};
+  if(id === undefined && tokenizerId === undefined) {
+    throw new TypeError('Either "id" or "tokenizerId" must be given.');
+  }
+  if(id !== undefined) {
+    if(!(Number.isInteger(id) && id >= 0)) {
+      throw new TypeError('"id" must be a non-negative integer.');
+    }
+    query['tokenVersion.id'] = id;
+  }
+  if(tokenizerId !== undefined) {
+    if(typeof tokenizerId !== 'string') {
+      throw new TypeError('"tokenizerId" must be a string.');
+    }
+    query['tokenVersion.tokenizerId'] = tokenizerId;
+  }
+  const projection = {_id: 0};
+  const collection = database.collections['tokenization-tokenVersion'];
+  const record = await collection.findOne(query, projection);
+  if(!record) {
+    const details = {
+      httpStatusCode: 404,
+      public: true
+    };
+    throw new BedrockError(
+      'Token version not found.',
+      'NotFoundError', details);
+  }
+  return record;
+}
+
+export async function setOptions({options}) {
+  const collection = database.collections['tokenization-tokenVersionOptions'];
+  const now = Date.now();
+  const tokenVersionOptions = {
+    id: TOKEN_OPTIONS_ID,
+    options
+  };
+  const query = {'tokenVersionOptions.id': TOKEN_OPTIONS_ID};
+  const $set = {tokenVersionOptions, 'meta.updated': now};
+  const upsertOptions = {...database.writeOptions, upsert: true};
+  const result = await collection.updateOne(query, {
+    $set,
+    $setOnInsert: {'meta.created': now}
+  }, upsertOptions);
+  if(result.result.n > 0) {
+    // upserted or modified
+    return true;
+  }
+  // no update, options were identical
+  return false;
+}
+
+export async function getOptions() {
+  const query = {'tokenVersionOptions.id': TOKEN_OPTIONS_ID};
+  const projection = {_id: 0};
+  const collection = database.collections['tokenization-tokenVersionOptions'];
+  const record = await collection.findOne(query, projection);
+  if(!record) {
+    const details = {
+      httpStatusCode: 404,
+      public: true
+    };
+    throw new BedrockError(
+      'Token version options not found.',
+      'NotFoundError', details);
+  }
+  return record;
+}
+
+export async function insertOptions({options}) {
+  const collection = database.collections['tokenization-tokenVersionOptions'];
+  const now = Date.now();
+  const meta = {created: now, updated: now};
+  let record = {
+    meta,
+    tokenVersionOptions: {
+      id: TOKEN_OPTIONS_ID,
+      options
+    }
+  };
+  try {
+    const result = await collection.insertOne(record, database.writeOptions);
+    record = result.ops[0];
+  } catch(e) {
+    if(!database.isDuplicateError(e)) {
+      throw e;
+    }
+    throw new BedrockError(
+      'Duplicate token version options.',
+      'DuplicateError', {
+        public: true,
+        httpStatusCode: 409
+      }, e);
+  }
+  return record;
+}
+
+async function _getNextVersionId() {
+  const projection = {_id: 0, 'tokenVersion.id': 1};
```

Comment:
Can either exclude or include fields, not both:
```suggestion
  const projection = {'tokenVersion.id': 1};
```

Reply:
http://mongodb.github.io/node-mongodb-native/3.5/tutorials/projections/

There is an exception to the rule which is the `_id` field.  The existing API call is proper.

Reply:
I should say the projection itself is OK, it *does* need to be passed into the options param as `{projection}` as is being suggested in other comments.